src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java

```diff
@@ -99,13 +106,22 @@ public static LocalDateTime getFirstDateTimeMustBe() {
         return null;
     }
 
-    public static int getHearingDuration(SscsCaseData caseData) {
+    public static int getHearingDuration(SscsCaseData caseData, ReferenceData referenceData) {
         // TODO Adjournments - Check this is the correct logic for Adjournments
         // TODO Future Work - Manual Override
-        // TODO Dependant on SSCS-10116 - Will use Session Category Reference Data
 
+        Integer duration = getHearingDurationAdjournment(caseData);
+        if (isNull(duration)) {
+            duration = getHearingDurationBenefitIssueCodes(caseData, referenceData);
+            if (isNull(duration)) {
+                duration = DURATION_DEFAULT;
+            }
+        }
```

Comment:
Nested if's are undesireable

Reply:
Suggest ternary operator would be appropriate here?